impala/tests/test_hs2_fault_injection.py

```diff
@@ -0,0 +1,357 @@
+        con.close()
+        assert self.__expect_msg_retry_with_retry_after("OpenSession") in caplog.text
+        assert self.__expect_msg_retry_with_retry_after_sleep() in caplog.text
+
+    def test_connect_proxy_retry_no_body(self, caplog):
+        """Tests fault injection in connect().
+        The injected error has no body but does have a Retry-After header that can be decoded.
+        Retries results in a successful connection."""
+        caplog.set_level(logging.DEBUG)
+        self.transport.enable_fault(503, "Injected Fault", 0.20, None,
+                                    {"header1": "value1",
+                                     "Retry-After": "1"})
+        con = self.connect()
+        cur = con.cursor(configuration=self.configuration)
+        cur.close()
+        con.close()
+        assert self.__expect_msg_retry_with_retry_after_no_extra("OpenSession") in caplog.text
+
+    def test_execute_query(self, caplog):
+        """Tests fault injection in execute().
+        ExecuteStatement rpc fails and results in error since retries are not supported."""
+        con = self.connect()
+        cur = con.cursor(configuration=self.configuration)
+        caplog.set_level(logging.DEBUG)
+        self.transport.enable_fault(502, "Injected Fault", 0.50)
+
+        query_handle = None
+        try:
+            query_handle = cur.execute('select 1')
+            assert False, 'execute should have failed'
+        except HttpError as e:
+            assert str(e) == 'HTTP code 502: Injected Fault'
+        assert query_handle is None
+        cur.close()
+        con.close()
+        assert self.__expect_msg_no_retry("ExecuteStatement") in caplog.text
+
+    def test_get_query_state(self, caplog):
+        """Tests fault injection in fetchall().
+        GetOperationStatus rpc fails and results in error since retries are not supported."""
+        con = self.connect()
+        cur = con.cursor(configuration=self.configuration)
+        caplog.set_level(logging.DEBUG)
+        cur.execute('select 1', {})
+        self.transport.enable_fault(502, "Injected Fault", 0.1)
+        cur.fetchall()
+        cur.close()
+        con.close()
+        assert self.__expect_msg_retry("GetOperationStatus") in caplog.text
+
+    def test_get_result_set_metadata(self, caplog):
+        """Tests fault injection in fetchcbatch().
+        GetResultSetMetadata rpc fails and is retried succesfully."""
+        con = self.connect()
+        cur = con.cursor(configuration=self.configuration)
+        caplog.set_level(logging.DEBUG)
+        cur.execute('select 1', {})
+        self.transport.enable_fault(502, "Injected Fault", 0.1)
+        cur.fetchcbatch()
+        cur.close()
+        con.close()
+        assert self.__expect_msg_retry("GetResultSetMetadata") in caplog.text
+
+    def test_fetch_results(self, caplog):
+        """Tests fault injection in fetchcbatch().
+        GetResultSetMetadata rpc fails and is retried succesfully."""
```

Comment:
The comment probably needs to be updated.

Reply:
Thanks